Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp

```diff
@@ -303,6 +320,11 @@ void MediaPlayerPrivateGStreamerMSE::updateStates()
             GST_ERROR_OBJECT(pipeline(), "Setting the pipeline to PAUSED failed");
         m_isPipelinePlaying = false;
     }
+
+    if (getStateResult == GST_STATE_CHANGE_SUCCESS && state >= GST_STATE_PAUSED) {
+        // Update playback rate that was set before the pipeline was prerolled.
+        updatePlaybackRate();
+    }
```

Comment:
I don't uniderstand this if. `getStateResult` and `state` hold the pipeline state previous to the change. PAUSED->PLAYING is asynchronous, PLAYING->PAUSED is not. That gives us the following possibilities:

* (1) When the function got called the state had been set to PAUSED (`getStateResult == GST_STATE_CHANGE_SUCCESS && state == GST_STATE_PAUSED`).
* <s>(2)</s> When the function got called the state had been set to PLAYING but the pipeline had not pre-rolled yet (`getStateResult == GST_STATE_CHANGE_ASYNC`). However, this case has been eliminated by a previous early return.
* (3) When the function got called the state had been set to PLAYING and the pipeline had already prerolled (`getStateResult == GST_STATE_CHANGE_SUCCESS && state == GST_STATE_PLAYING`).
* <s>(4)</s> Playback has ended with a state change to READY or NULL. I don't know of a case where this happens in the MSE player.

So this seems like a condition that would always be executed. I'm also confused because it checks the old state rather than the new but doesn't explain why, I don't get it.

Reply:
There are other state changes than success and async. I guess it makes sense to not do anything in any other than success. Then, if we have already a state change sinked, i.e. success, what we want is to call `updatePlaybackRate` so that it can move the pipeline to playing if the rate is suitable for that.

Reply:
> There are other state changes than success and async.

Although they exist in the abstract, I don't see how they could happen in this particular case, and if they can, it should be explained.